### PR TITLE
110 gravatar image source

### PR DESCRIPTION
**Pull request by GeorgeLeithead**
Added documentation for CommunityToolkit proposal [110-GravatarImageSource](https://github.com/CommunityToolkit/Maui/issues/110)

### CODE REVIEW

**Comment by PRMerger8** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Index File | This PR contains a change to the index file.

File: docs/maui/views/GravatarImageSource.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+```
+
+## Properties
+
+| Property | Type | Description |
+|---|---|---|
+| CacheValidity | `TimeSpan` | The `CacheValidity` property, of type `TimeSpan`, specifies how long the image will be stored locally for. The default value of this property is 1 day. |
+| CachingEnabled | `bool` | The `CachingEnabled` property, of type `bool`, defines whether image caching is enabled. The default value of this property is `true`. |
+| Email | `string?` | The `Email` property, of type `string?`, specifies the gravatar account email address.  If unset, the Gravatar image is rendered. If set and not found on Gravatar, the `Image` property image will be rendered. |
+| Image | `DefaultImage` | The `Image` property, of type `DefaultImage` is an enumeration that is used to specify the default image if the `email` is not found on Gravatar. |
```
Comment (bijington) [inline docs/maui/views/GravatarImageSource.md:66]: Perhaps we can link to the example below which shows the possible values for the `Image` property?

**Review by bijington — CHANGES_REQUESTED** [review]
@GeorgeLeithead thanks for this it looks great! The only change that we will need to do is move out of `views` as per the request on the proposal PR.

File: docs/maui/views/GravatarImageSource.md (hunk 1 above)
Comment (GeorgeLeithead) [reply docs/maui/views/GravatarImageSource.md]: Added link to example.

File: docs/maui/imagesources/index.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ms.date: 08/18/2022
+---
+
+# ImageSources
+
+The user interface of a .NET Multi-platform App UI (.NET MAUI) app is constructed of objects that map to the native controls of each target platform.
+
+The main control groups used to create the user interface of a .NET MAUI app are pages, layouts, and views. A .NET MAUI page generally occupies the full screen or window. The page usually contains a layout, which contains views and possibly other layouts. Pages, layouts, and views derive from the `VisualElement` class. This class provides a variety of properties, methods, and events that are useful in derived classes.
+
+For further information on Behaviors please refer to the [.NET MAUI documentation](/dotnet/maui/user-interface/controls/).
```
Comment (bijington) [inline docs/maui/imagesources/index.md:14]: Do we need this bit?

File: docs/maui/imagesources/index.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+The .NET Multi-platform App UI (.NET MAUI) Image displays an `image` that can be loaded from a local file, a URI, an embedded resource, or a stream. The standard platform image formats are supported, including animated GIFs, and local Scalable Vector Graphics (SVG) files are also supported.  For more information about the `Image` control, see [Image](/dotnet/maui/user-interface/controls/image).
+
+`Images` defin a `Source` property, of type `ImageSource`, which specifies the source of the image.  The `ImageSource` class defines the following methods that can be used to load an image from different sources:
```
Comment (bijington) [inline docs/maui/imagesources/index.md:18]: Spelling mistake

```suggestion
`Images` define a `Source` property, of type `ImageSource`, which specifies the source of the image.  The `ImageSource` class defines the following methods that can be used to load an image from different sources:
```

Comment (bijington) [reply docs/maui/imagesources/index.md:18]: Perhaps we could elude to there being more controls than just `Image`?

**Review by bijington — CHANGES_REQUESTED** [review]
Thanks @GeorgeLeithead I just noticed a few little bits but I think the bulk is good

File: docs/maui/imagesources/index.md (hunk 2 above)
Comment (GeorgeLeithead) [reply docs/maui/imagesources/index.md:14]: Removed.

File: docs/maui/imagesources/GravatarImageSource.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
+---
+title: GravatarImageSource - .NET MAUI Community Toolkit
+author: GeorgeLeithead
+description: "`GravatarImageSource` allows you to use as an Image source, a users Gravatar registered image via their email address."
+ms.date: 08/05/2022
```
Comment (jfversluis) [inline docs/maui/imagesources/GravatarImageSource.md:5]:
```suggestion
ms.date: 09/16/2022
```

File: docs/maui/imagesources/GravatarImageSource.md [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+    xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit">
+    <VerticalStackLayout>
+        <Image>
+            <Image.Source>
+                <toolkit:GravatarImageSource
+                    CacheValidity="1"
+                    CachingEnabled="True"
+                    Email="youremail@here"
```
Comment (jfversluis) [inline docs/maui/imagesources/GravatarImageSource.md:27]:
```suggestion
                    Email="youremail@here.com"
```

File: docs/maui/imagesources/GravatarImageSource.md [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+{
+	public MyPage()
+	{
+        Image myImage = new()
+        {
+            Source = new GravatarImageSource()
+            {
+                CacheValidity = TimeSpan.FromDays(1),
+                CachingEnabled = true,
+                Email = "youremail@here",
```
Comment (jfversluis) [inline docs/maui/imagesources/GravatarImageSource.md:50]:
```suggestion
                Email = "youremail@here.com",
```

File: docs/maui/imagesources/GravatarImageSource.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+};
+```
+
+The following example sets an email address that has no matching Gravatar image and will thus display the default `Image` image.
+
+
+```xaml
+<Image>
+    <Image.Source>
+        <toolkit:GravatarImageSource Email="notregistered@emailongravitar" />
```
Comment (jfversluis) [inline docs/maui/imagesources/GravatarImageSource.md:174]:
```suggestion
        <toolkit:GravatarImageSource Email="notregistered@emailongravitar.com" />
```

File: docs/maui/imagesources/GravatarImageSource.md [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+```
+
+The equivalent C# code is:
+
+```csharp
+Image myImage = new()
+{
+    Source = new GravatarImageSource()
+    {
+        Email = "notregistered@emailongravitar",
```
Comment (jfversluis) [inline docs/maui/imagesources/GravatarImageSource.md:186]:
```suggestion
        Email = "notregistered@emailongravitar.com",
```

File: docs/maui/imagesources/GravatarImageSource.md [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+The `Image` property is an enumeration that is used to specify the default image if the `email` address has no matching Gravatar image.  The available options are:
+
+- MysteryPerson (default) - A simple, cartoon-style silhouetted outline of a person (does not vary by email hash)
+- FileNotFound - Do not load any image if none is associated with the email hash, instead return an HTTP 404 (File Not Found response.
+- Identicon - A geometric pattern based on an email hash.
+- MonsterId - A generated 'monster' with different colours, faces, etc.
+- Wavatar - Generated faces with differing features and backgrounds.
+- Retro - Awesome generated, 8-bit arcade-style pixilated faces.
+- Robohash - A generated robot with different colours, faces, etc.
+- Blank - A transparent PNG image.
```
Comment (jfversluis) [inline docs/maui/imagesources/GravatarImageSource.md:201]:
```suggestion
- `MysteryPerson` (default) - A simple, cartoon-style silhouetted outline of a person (does not vary by email hash)
- `FileNotFound` - Do not load any image if none is associated with the email hash, instead return an HTTP 404 (File Not Found) response.
- `Identicon` - A geometric pattern based on an email hash.
- `MonsterId` - A generated 'monster' with different colours, faces, etc.
- `Wavatar` - Generated faces with differing features and backgrounds.
- `Retro` - Awesome generated, 8-bit arcade-style pixilated faces.
- `Robohash` - A generated robot with different colours, faces, etc.
- `Blank` - A transparent PNG image.
```

File: docs/maui/imagesources/GravatarImageSource.md [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+
+The following example sets the default image of a `GravatarImageSource`:
+
+```xaml
+<Image>
+    <Image.Source>
+        <toolkit:GravatarImageSource Email="notregistered@emailongravitar" Image="Retro" />
```
Comment (jfversluis) [inline docs/maui/imagesources/GravatarImageSource.md:208]:
```suggestion
        <toolkit:GravatarImageSource Email="notregistered@emailongravitar.com" Image="Retro" />
```

File: docs/maui/imagesources/GravatarImageSource.md [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+```
+
+The equivalent C# code is:
+
+```csharp
+Image myImage = new()
+{
+    Source = new GravatarImageSource()
+    {
+        Email = "notregistered@emailongravitar",
```
Comment (jfversluis) [inline docs/maui/imagesources/GravatarImageSource.md:220]:
```suggestion
        Email = "notregistered@emailongravitar.com",
```

File: docs/maui/imagesources/index.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
+---
+title: ImageSources - .NET MAUI Community Toolkit
+author: GeorgeLeithead
+description: The .NET MAUI Community Toolkit extends .NET MAUI ImageSources.
+ms.date: 08/18/2022
```
Comment (jfversluis) [inline docs/maui/imagesources/index.md:5]:
```suggestion
ms.date: 09/16/2022
```

File: docs/maui/imagesources/index.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+---
+
+# ImageSources
+
+The .NET Multi-platform App UI (.NET MAUI) Image displays an `image` that can be loaded from a local file, a URI, an embedded resource, or a stream. The standard platform image formats are supported, including animated GIFs, and local Scalable Vector Graphics (SVG) files are also supported.  For more information about the `Image` control, see [Image](/dotnet/maui/user-interface/controls/image).
```
Comment (jfversluis) [inline docs/maui/imagesources/index.md:10]:
```suggestion
The .NET Multi-platform App UI (.NET MAUI) `Image` displays an image that can be loaded from a local file, a URI, an embedded resource, or a stream. The standard platform image formats are supported, including animated GIFs, and local Scalable Vector Graphics (SVG) files are also supported.  For more information about the `Image` control, see [Image](/dotnet/maui/user-interface/controls/image).
```

**Review by jfversluis — CHANGES_REQUESTED** [review]

**Review by jfversluis — APPROVED** [review]

**Review by bijington — APPROVED** [review]

**Pull request merged**